doc/rm/c_cpp_coding_style.md

```diff
@@ -139,6 +139,31 @@ Example:
 int create_rainbow(int pots_of_gold, int unicorns);
 ```
 
+### Polyglot headers
+
+***Headers intended to be included from both languages must contain `extern` guards; `#include`s should not be wrapped in `extern "C" {}`.***
+
+A *polyglot header* is a header file that can be safely included in either a `.c` or `.cc` file.
+In particular, this means that the file does not depend on any of the places where C and C++ semantics disagree.
+
+Such files must be explictly marked with `extern` guards like the following, starting after the file's `#include`s.
+```
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Polyglot declarations...
+
+#ifdef __cplusplus
+}  // extern "C"
+#endif
+```
+Moreover, all `#includes` must also be of other polyglot headers.
```

Comment:
Doesn't this force considerations on other header writers after the fact? If you're including a header that wasn't intended to be polyglot (which is allowed from the third line of this change) then you don't have to write `extern` decls.

I think it might be easier to just make a requirement that headers for C libraries must carry `extern C` declarations.

Reply:
Sorry, I think I wasn't precise in my wording. Here's the intended meaning:
- Headers *may* be polyglot.
- A polyglot header *must* only include other polyglot headers.
- A polyglot header's body, after the includes, must be wrapped in `extern "C"`.

Exactly what did I write that is inconsistent with that? What can I add to clarify?